qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -57,3 +61,50 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown sample template action: %s' % action)
         self.write(result)
+
+
+def build_sample_summary(study_id, user_id, visible=None):
```

Comment:
I'm concerned about the performance of this call. This is going over the entire sample template and all the prep templates on that study. In cases like the AGP, this can become _really_ expensive.

Why has been all the metadata summary merged in a single page? Before it was per each template. I think that was the correct way of doing it...